File: A. Easy/0414-third-maximum-number/0414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        Max1 = Max2 = Max3 = -(2**31) - 1

        for num in nums:
            if((num == Max1) or (num == Max2) or (num == Max3)):
                continue
            
            if (num > Max1):
                Max3 = Max2
                Max2 = Max1
                Max1 = num
            elif (num > Max2):
                Max3 = Max2
                Max2 = num
            elif (num > Max3):
                Max3 = num
        
        if (Max3 == -(2**31) - 1):
            return Max1
        return Max3
        
        

File: Difficulty/A. Easy/0414-third-maximum-number/0414-third-maximum-number.c
int thirdMax(int* nums, int numsSize) {
    long Max1 = LONG_MIN;
    long Max2 = LONG_MIN;
    long Max3 = LONG_MIN;

    for (int i = 0; i < numsSize; i++) {
        if ((nums[i] == Max1) || (nums[i] == Max2) || (nums[i] == Max3)) {
            continue;
        }

        if (nums[i] > Max1) {
            Max3 = Max2;
            Max2 = Max1;
            Max1 = nums[i];
        }
        else if (nums[i] > Max2) {
            Max3 = Max2;
            Max2 = nums[i];
        }
        else if (nums[i] > Max3) {
            Max3 = nums[i];
        }
    }
    
    if (Max3 == LONG_MIN) {
        return (int)Max1;
    }
    return (int)Max3;
}